### ISSUE TITLE
Not clear which typescript versions are supported

### ISSUE THREAD

**Issue by elycruz** [lit/lit#2731]
## Description

There is a need to know what typescript versions are supported by the project since legacy typescript projects (Angular 8 etc.) use typescript versions that don't support the *.d.ts files generated in the latest 'lit' library (including 'lit-element', and, 'lit-html' libraries).

Typescript incompatibility described in #2730 , #2644 .

# Steps to Reproduce, and Additional Details

See issues #2730 , #2644 .

**Comment by justinfagnani** [comment]
Thanks for raising this. We do need to communicate our support level here and potentially setup downlevel-dts to ensure base-level support.

**Comment by elycruz** [comment]
Yeah, not a prob. (👍🏽 ).  Here's what I did in my client project:

For testing:

Tested client library in applications running on:

- Angular 13.3.x (requires typescript `~4.6.2` (https://github.com/angular/angular/blob/41c6f7a0796193728981572c8bef900b23da6c3f/package.json#L155) , and
- Angular 8 (requires typescript `~3.5.3` (https://github.com/angular/angular/blob/6e30b83267cdb4be62cc2c9737aeac62d14ed513/package.json#L111)

For configuration: 

In package.json (all non-relevant details left out) followed instructions in downlevel-dts's README (https://www.npmjs.com/package/downlevel-dts), and made sure my 'downleveled' *.d.ts files were included in my (npm) package deployment;  E.g.

```json
{
"typesVersions": {
    "<4.5": {
      "*": [
        "ts3.5/*"
      ]
    }
  },
  "scripts": {
    "downlevel-dts": "npx downlevel-dts . ts3.5 [--to=3.5]"
  },
  "devDependencies": {
    "downlevel-dts": "^0.8.0",
  }
}
```

Ensure target downleveled *.d.ts files directory is included in (npm) package deployment (in this scenario that's '{repo-root}/ts.3.5/').
